src/lib/annotations/doc/DocAnnotator.js

```diff
@@ -391,7 +432,19 @@ class DocAnnotator extends Annotator {
         // Scale existing canvases on re-render
         this.scaleAnnotationCanvases(pageNum);
 
-        super.renderAnnotationsOnPage(pageNum);
+        if (!this.threads) {
+            return;
+        }
+
+        // TODO (@jholdstock|@spramod) remove this if statement, and make super call, upon refactor.
+        const pageThreads = this.getThreadsOnPage(pageNum);
```

Comment:
at this point, isn't this the same as calling `super.renderAnnotationsOnPage()`? Since that also checks `isModeAnnotatable(type)` and that's essentially when `this.plainHighlightEnabled` and `this.commentHighlightEnabled` is?

Reply:
Note that the override is passing in the two flags, which are **not** done in the parent class
```
 thread.show(this.plainHighlightEnabled, this.commentHighlightEnabled);
```